### DISCUSSION TITLE
Active clustering of faces with user feedback

### DISCUSSION THREAD

**Discussion by vishnukvmd** [ente-io/ente#549] [Enhancements]

**Comment by ridicolos** [comment]
I tried the face recognition feature in the desktop environment on two devices with the same photo library. What I found a bit confusing is that on the Mac, my girlfriend was recognized as three different people. On Windows on the other side she was recognized as two different people.

So it would be cool to be able to merge multiple people into one.

That the info on the already indexed images are not synronized is certainly because the whole thing is still in beta, right? 

Other ideas and features:

- Also, it would be pretty cool if like on iOS, you could also give that person a name. Then I could search only for the person "dennis" and would get all the photos on which I can be seen.

- To set a custom thumbnail for the person would be a great enhancement. Duck has really not chosen the most beautiful photos for the thumbnail.

- I know that ente currently does not support nested albums, but that could be a cool feature, which you could use for the person-overview. So you would have something like a folder called `people` and in there you have a tile for all of the indexed persons. Then you could easly set options like merging them together, giving them names, settings a custom thumbnail and so on.

**Comment by laurenspriem** [comment]
Thanks for your thoughts @ridicolos always good to have community feedback!

We are already actively working on creating some mechanisms for the user to give feedback and improve the people clustering. I'm currently working on this for the mobile app, but this will also be implemented on web. And indeed, once we release this (hopefully very soon) we will also sync the information on the indexed images to other devices.

Now for your suggestions, in order:

1. Yes we will give you the option to name clusters and search on names
2. Yes there will be an option to change the face thumbnail of a cluster
3. I'm not sure if I completely follow, indeed ente doesn't support nested albums. What I can say is that we're going to completely revamp the search screen, which will be a separate tab like albums and the sharing tab. We're not sure yet what the best UX/UI is for the feedback, but I can imagine we will put some of the feedback features here. See our figma for the current design we had in mind: https://www.figma.com/file/SYtMyLBs5SAOkTbfMMzhqt/ente-Visual-Design?type=design&node-id=15927-66193&mode=design&t=TSg4OxmeFDDF4x0f-4

To give full transparency on what we're working on, these are the feedback mechanisms we are currently working on:
1. Remove photo from a cluster
2. Add photo to a cluster
4. Merge two (or more) clusters
5. Delete a cluster
6. Name a cluster
7. Change thumbnail of cluster

We understand how to implement these features under the hood, however we don't have full clarity on what the UX/UI for these will look like. So if you have any thoughts on that, please do share! Also, if there's anything else you feel is missing or could be done differently, let us know :)
